src/dfx/src/lib/operations/canister/create_canister.rs

```diff
@@ -42,7 +42,8 @@ pub async fn create_canister(
 
     if let Some(remote_canister_id) = config
         .get_config()
-        .get_remote_canister_id(canister_name, &network_name)?
+        .get_remote_canister_id(canister_name, &network_name)
+        .unwrap_or_default()
```

Comment:
If creating a canister not in dfx.json, this would bail.  Now, when checking whether there is already a defined remote canister ID, we don't care if there is no canister entry whatsoever in dfx.json.